spring-kafka/src/main/java/org/springframework/kafka/config/ShareKafkaListenerContainerFactory.java

```diff
@@ -0,0 +1,240 @@
+		}
+		if (this.containerProperties.getAckTime() > 0) {
+			properties.setAckTime(this.containerProperties.getAckTime());
+		}
+		if (endpoint.getAutoStartup() != null) {
+			instance.setAutoStartup(endpoint.getAutoStartup());
+		}
+		else if (this.autoStartup != null) {
+			instance.setAutoStartup(this.autoStartup);
+		}
+		if (this.phase != null) {
+			instance.setPhase(this.phase);
+		}
+		if (this.applicationContext != null) {
+			instance.setApplicationContext(this.applicationContext);
+		}
+		if (this.applicationEventPublisher != null) {
+			instance.setApplicationEventPublisher(this.applicationEventPublisher);
+		}
+		if (endpoint.getGroupId() != null) {
+			instance.getContainerProperties().setGroupId(endpoint.getGroupId());
+		}
+		if (endpoint.getClientIdPrefix() != null) {
+			instance.getContainerProperties().setClientId(endpoint.getClientIdPrefix());
+		}
+		if (endpoint.getConsumerProperties() != null) {
+			instance.getContainerProperties().setKafkaConsumerProperties(endpoint.getConsumerProperties());
+		}
+	}
+
+	@Override
+	public ShareKafkaMessageListenerContainer<K, V> createContainer(TopicPartitionOffset... topicPartitions) {
+		return createContainerInstance(new KafkaListenerEndpointAdapter() {
+			@Override
+			public TopicPartitionOffset[] getTopicPartitionsToAssign() {
+				return Arrays.copyOf(topicPartitions, topicPartitions.length);
+			}
+		});
+	}
+
+	@Override
+	public ShareKafkaMessageListenerContainer<K, V> createContainer(String... topics) {
+		return createContainerInstance(new KafkaListenerEndpointAdapter() {
+			@Override
+			public Collection<String> getTopics() {
+				return Arrays.asList(topics);
+			}
+		});
+	}
+
+	@Override
+	public ShareKafkaMessageListenerContainer<K, V> createContainer(Pattern topicPattern) {
+		return createContainerInstance(new KafkaListenerEndpointAdapter() {
+			@Override
+			public Pattern getTopicPattern() {
+				return topicPattern;
+			}
+		});
+	}
+
+	/**
+	 * Create a container instance for the provided endpoint.
+	 * @param endpoint the endpoint
+	 * @return the container instance
+	 */
+	protected ShareKafkaMessageListenerContainer<K, V> createContainerInstance(KafkaListenerEndpoint endpoint) {
+		TopicPartitionOffset[] topicPartitions = endpoint.getTopicPartitionsToAssign();
+		if (topicPartitions != null && topicPartitions.length > 0) {
+			return new ShareKafkaMessageListenerContainer<>(getShareConsumerFactory(), new ContainerProperties(topicPartitions));
+		}
+		else {
+			Collection<String> topics = endpoint.getTopics();
+			Assert.state(topics != null, "'topics' must not be null");
+			if (!topics.isEmpty()) {
+				return new ShareKafkaMessageListenerContainer<>(getShareConsumerFactory(),
+						new ContainerProperties(topics.toArray(new String[0])));
+			}
+			else {
+				return new ShareKafkaMessageListenerContainer<>(getShareConsumerFactory(),
+						new ContainerProperties(endpoint.getTopicPattern()));
```

Comment:
Probably `KafkaShareConsumer` does not support pattern subscription as well...
Please, revise.